### PR TITLE
Fixes/mobile bugs hardcode bugs

### PR DESCRIPTION
**Pull request by W3stside**
Fixes bugs:

Mobile: `MenuAuctions` absolute positioning + `left: 0` with new container was hiding container outside of screen. 

Dynamic text to replace hardcoded `MetaMask` provider name in `WalletPanel` screen (just before blockchain modals)

Attempts to fix `noScroll` and scroll behaviour on modals especially on `Final COnfirmation` screen that has a lot of content on mobile. Previous changes was showing non-scrollable Y scrollbar due to `overflow-y: scroll` on at all times

and other stuff....

### CODE REVIEW

File: src/components/App/index.tsx [hunk 1]
```diff
@@ -38,4 +38,9 @@ const App = (props: AppProps): any => {
     </Provider>
   )}
 
+// history listen on change
+history.listen((loc: any) => {
+  loc.search.includes('retrox') && document.body.classList.add('RETROX')
```
Comment (Velenir) [inline src/components/App/index.tsx]: I though about it and a better, more compatible way would be
```js
const searchParams = new URLSearchParams(location.search)
document.body.classList.toggle('RETOX', searchParams.has('retox'))
```

File: src/components/Hamburger/index.tsx [hunk 2]
```diff
@@ -37,7 +37,7 @@ export default class Hamburger extends Component<HamburgerProps, HamburgerState>
             className="buttonExit"
             onClick={this.handleClick}>
           </button>
-          {window && window.location.pathname !== '/' && <Link to="/" onClick={this.handleClick}> Home </Link>}
+          {window && window.location.hash !== '#/' && <Link to="/" onClick={this.handleClick}> Home </Link>}
```
Comment (Velenir) [inline src/components/Hamburger/index.tsx]: Would be better not to depend on window here, as that stuff will change based on Hash/Browser history every time
Try
```
<Route exact path="/" render={() => <Link to="/" onClick={this.handleClick}> Home </Link>}/>
```

File: src/middlewares/NoScroll.ts [hunk 3]
```diff
@@ -1,17 +1,16 @@
...
 const NoScroll = () => (next: Function) => async (action: any) => {
   const { type } = action as Action
-
-  if (action.payload && action.payload.modalProps && action.payload.modalProps.txData || type !== 'CLOSE_MODAL' && type !== 'OPEN_MODAL') return next(action as Action)
+  if (action.payload && action.payload.modalProps && action.payload.modalProps.header !== 'Order confirmation' || type !== 'CLOSE_MODAL' && type !== 'OPEN_MODAL') return next(action as Action)
 
   try {
     if (type === 'OPEN_MODAL') {
       document && document.body && document.body.classList.add('noScroll')
+      document && document.getElementById('root') && document.getElementById('root').classList.add('noScroll')
```
Comment (Velenir) [inline src/middlewares/NoScroll.ts]: Don't search for element twice unnecessarily
```
const root = document && document.getElementById('root')
root && root.classList.add('noScroll')
```

File: src/middlewares/NoScroll.ts [hunk 4]
```diff
@@ -1,17 +1,16 @@
...
     } else if (type === 'CLOSE_MODAL') {
       document && document.body && document.body.classList.remove('noScroll')
+      document && document.getElementById('root') && document.getElementById('root').classList.remove('noScroll')
```
Comment (Velenir) [inline src/middlewares/NoScroll.ts]: same as above

**Review by Velenir — COMMENTED** [review]
Wondering if we can cut down on it by using css vars.
Let's speak Monday

File: src/components/Hamburger/index.tsx (hunk 2 above)
Comment (W3stside) [reply src/components/Hamburger/index.tsx]: but this doesn't make sense, render Home when on "/" aka home path? should be the inverse

File: src/components/App/index.tsx (hunk 1 above)
Comment (W3stside) [reply src/components/App/index.tsx]: works well!

File: src/components/Hamburger/index.tsx (hunk 2 above)
Comment (Velenir) [reply src/components/Hamburger/index.tsx]: Yes, a million times
```
<Route path="/(.+)" render={() => <Link to="/" onClick={this.handleClick}> Home </Link>}/>
```
this works

Comment (W3stside) [reply src/components/Hamburger/index.tsx]: cool, good stuff ;)

**Review by Velenir — APPROVED** [review]

**Pull request merged**